### PR TITLE
Bind fixes

### PR DESCRIPTION
**Pull request by rcorre**

### CODE REVIEW

File: qutebrowser/commands/command.py [hunk 1]
```diff
@@ -101,10 +100,14 @@ def __init__(self, *, handler, name, instance=None, maxsplit=None,
             for m in modes:
                 if not isinstance(m, usertypes.KeyMode):
                     raise TypeError("Mode {} is no KeyMode member!".format(m))
-        if not_modes is not None:
+            self._modes = modes
+        elif not_modes is not None:
             for m in not_modes:
                 if not isinstance(m, usertypes.KeyMode):
                     raise TypeError("Mode {} is no KeyMode member!".format(m))
+            self._modes = [m for m in usertypes.KeyMode if m not in not_modes]
```
Comment (The-Compiler) [inline qutebrowser/commands/command.py]: Why not make `self._modes` a set? Then we could simply do `set(usertypes.KeyMode).difference(not_modes)` if I'm not mistaken.

File: qutebrowser/commands/command.py [hunk 2]
```diff
@@ -525,3 +516,15 @@ def run(self, win_id, args=None, count=None):
         log.commands.debug('Calling {}'.format(
             debug_utils.format_call(self.handler, posargs, kwargs)))
         self.handler(*posargs, **kwargs)
+
+    def validate_mode(self, mode):
```
Comment (The-Compiler) [inline qutebrowser/commands/command.py]: This should be `_validate_mode` to make it "private"

File: qutebrowser/commands/command.py [hunk 3]
```diff
@@ -525,3 +516,15 @@ def run(self, win_id, args=None, count=None):
...
+        """Raise cmdexc.PrerequisitesError unless allowed in the given mode.
+
+        Args:
+            mode: The usertypes.KeyMode to check.
+        """
+        if mode not in self._modes:
+            mode_names = '/'.join(m.name for m in self._modes)
```
Comment (The-Compiler) [inline qutebrowser/commands/command.py]: If we make `self._modes` a set we'll probably need a `sorted()` here.

File: qutebrowser/commands/command.py (hunk 2 above)
Comment (rcorre) [inline qutebrowser/commands/command.py]: I call `validate_mode` from `bind` in `keyconf`

Comment (The-Compiler) [inline qutebrowser/commands/command.py]: Oh, right - sorry, missed that!

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**